includes/Handlers/Connection.php

```diff
@@ -0,0 +1,303 @@
+
+defined( 'ABSPATH' ) or exit;
+
+/**
+ * The connection handler.
+ *
+ * @since 2.0.0-dev.1
+ */
+class Connection {
+
+
+	/** @var string the action callback for the connection */
+	const ACTION_CONNECT = 'wc_facebook_connect';
+
+	/** @var string the WordPress option name where the external business ID is stored */
+	const OPTION_EXTERNAL_BUSINESS_ID = 'wc_facebook_external_business_id';
+
+	/** @var string the business manager ID option name */
+	const OPTION_BUSINESS_MANAGER_ID = 'wc_facebook_business_manager_id';
+
+	/** @var string the access token option name */
+	const OPTION_ACCESS_TOKEN = 'wc_facebook_access_token';
+
+
+	/** @var string|null the generated external merchant settings ID */
+	private $external_business_id;
+
+
+	/**
+	 * Constructs a new Connection.
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function __construct() {
+
+	}
+
+
+	/**
+	 * Processes the returned connection.
+	 *
+	 * @internal
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function handle_connect() {
+
+	}
+
+
+	/**
+	 * Disconnects the integration using the Graph API.
+	 *
+	 * @internal
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function handle_disconnect() {
+
+	}
+
+
+	/**
+	 * Converts a temporary user token to a system user token via the Graph API.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param string $user_token
+	 * @return string
+	 */
+	public function create_system_user_token( $user_token ) {
+
+		return $user_token;
+	}
+
+
+	/**
+	 * Gets the API access token.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @return string
+	 */
+	public function get_access_token() {
+
+		return get_option( self::OPTION_ACCESS_TOKEN, '' );
+	}
+
+
+	/**
+	 * Gets the URL to start the connection flow.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @return string
+	 */
+	public function get_connect_url() {
+
+		return '';
+	}
+
+
+	/**
+	 * Gets the URL for disconnecting.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @return string
+	 */
+	public function get_disconnect_url() {
+
+		return '';
+	}
+
+
+	/**
+	 * Gets the scopes that will be requested during the connection flow.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @return string[]
+	 */
+	public function get_scopes() {
+
+		$scopes = [
+			'manage_business_extension',
+			'catalog_management',
+			'business_management',
+		];
+
+		/**
+		 * Filters the scopes that will be requested during the connection flow.
+		 *
+		 * @since 2.0.0-dev.1
+		 *
+		 * @param string[] $scopes connection scopes
+		 * @param Connection $connection connection handler instance
+		 */
+		return (array) apply_filters( 'wc_facebook_connection_scopes', $scopes, $this );
+	}
+
+
+	/**
+	 * Gets the stored external business ID.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @return string
+	 */
+	public function get_external_business_id() {
+
+		if ( ! is_string( $this->external_business_id ) ) {
+
+			$value = get_option( self::OPTION_EXTERNAL_BUSINESS_ID );
+
+			if ( ! is_string( $value ) ) {
+
+				$value = sanitize_title( get_bloginfo( 'name' ) ) . '-' . uniqid();
```

Comment:
```suggestion
				$value = sanitize_title( get_bloginfo( 'name' ) ) . '-' . uniqid( '', true );
```

Reply:
not sure if uniqid here should have 2nd param as true

Reply:
I'm not sure if the `.` that is added to the unique ID is allowed. What about:
```suggestion
				$value = sanitize_title( uniqid( get_bloginfo( 'name' ) . '-', true ) )
```

That should give something like:

    skyverge-plugins-5ebbeab9843e86-78530995